File: src/application/module/brs/services/BlockService.java
package application.module.brs.services;

import application.module.brs.Block;
import application.module.brs.BlockchainProcessor;
import application.module.brs.BlockchainProcessor.BlockNotAcceptedException;
import application.module.brs.BlockchainProcessor.BlockOutOfOrderException;

public interface BlockService {

    void preVerify(Block block, Block prevBlock)
            throws BlockchainProcessor.BlockNotAcceptedException, InterruptedException;

    void preVerify(Block block, Block prevBlock, byte[] scoopData)
            throws BlockchainProcessor.BlockNotAcceptedException, InterruptedException;

    void watchBlock(Block block);

    long getBlockReward(Block block);

    void calculateBaseTarget(Block block, Block lastBlock) throws BlockOutOfOrderException;

    void setPrevious(Block block, Block previousBlock);

    boolean verifyGenerationSignature(Block block) throws BlockNotAcceptedException;

    boolean verifyBlockSignature(Block block) throws BlockOutOfOrderException;

    void apply(Block block);

    int getScoopNum(Block block);
}

File: src/main/kotlin/brs/services/BlockService.kt
package brs.services

import brs.entity.Block

interface BlockService {
    @Throws(BlockchainProcessorService.BlockNotAcceptedException::class, InterruptedException::class)
    fun preVerify(block: Block)

    @Throws(BlockchainProcessorService.BlockNotAcceptedException::class, InterruptedException::class)
    fun preVerify(block: Block, scoopData: ByteArray?)

    fun getBlockReward(block: Block): Long

    fun calculateBaseTarget(block: Block, previousBlock: Block)

    fun setPrevious(block: Block, previousBlock: Block?)

    fun verifyGenerationSignature(block: Block): Boolean

    fun verifyBlockSignature(block: Block): Boolean

    fun apply(block: Block)

    fun getScoopNum(block: Block): Int
}
